providers/base/bin/network_device_info.py

```diff
@@ -82,14 +80,14 @@ def is_iface_connected(cls, iface):
     def get_ipv4_address(interface):
         s = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
         try:
-            ipv4_addr = socket.inet_ntoa(fcntl.ioctl(
-                s.fileno(),
-                0x8915,  # SIOCGIFADDR
-                struct.pack('256s', interface[:15].encode())
-            )[20:24])
-        except Exception as e:
-            print("ERROR: getting the IPv4 address for %s: %s" %
-                  (interface, repr(e)))
+
+            buffer = struct.pack('256s', interface[:15].encode())
+            # retrieve the IP address associated with a network interface.
+            data = fcntl.ioctl(s.fileno(), 0x8915, buffer)
+            ipv4_addr = socket.inet_ntoa(data[20:24])
+        except OSError:
+            print("OSError: failed to get the IPv4 address for %s" %
+                  interface, file=sys.stderr)
```

Comment:
using `file=sys.stderr` is a good idea! It will indeed show up as a red message in the Checkbox output.

I would keep the error message from the exception (`repr(e)`) displayed, because it will make debugging easier.